invenio_records_lom/services/schemas/metadata.py

```diff
@@ -56,27 +74,77 @@ def _deserialize(self, value, attr, data, **kwargs):
 
         # validate inner keys
         if not isinstance(langstring_inner, dict):
-            raise self.make_error("invalid")
+            raise self.make_error("invalid_inner_type")
         inner_keys = set(langstring_inner)
         if "#text" not in inner_keys:
             raise self.make_error("missing_langstring_key", keys="#text")
-        if "lang" not in inner_keys and self.lang_exists:
+        if "lang" not in inner_keys and self.lang_exists is True:
             raise self.make_error("missing_langstring_key", keys="lang")
-        if "lang" in inner_keys and not self.lang_exists:
+        if "lang" in inner_keys and self.lang_exists is False:
             raise self.make_error("extraneous_lang")
         if not inner_keys <= {"#text", "lang"}:
             raise self.make_error(
                 "extraneous_keys", keys=sorted(inner_keys - {"#text", "lang"})
             )
 
-        text = langstring_inner["#text"]
-        lang = langstring_inner.get("lang", True)
+        text = sanitize_unicode(str(langstring_inner["#text"]))
+        lang_exists = "lang" in langstring_inner
+        lang = sanitize_unicode(str(langstring_inner["lang"])) if lang_exists else None
 
         # validate non-emptiness of "#text"-, "lang"-key
-        if not text or not lang:
+        if not text:
+            raise self.make_error("required")
+        if not lang_exists and self.lang_exists is True:
             raise self.make_error("required")
 
-        return value
+        # validation succeded, rebuild langstring with sanitized text/lang
+        result = {"langstring": {"#text": text}}
+        if lang_exists:
+            result["langstring"]["lang"] = lang
+
+        return result
+
+
+def validate_langstring_lang(validator_callable):
+    """Wraps `validator_callable`, passing it value["langstring"]["lang"] instead of value itself."""
+
+    def validate_lang(value):
+        try:
+            value = value["langstring"]["lang"]
```

Comment:
style: don't change type of variable. `value` was as the parameter a `dict`, after this assignment it is a `str`. i know it is possible in python, but please don't do it.

i think `lang` would be the better name for the variable. this helps also to see that the value given to `validator_callable` is not the same as the `validate_lang` function received.

Reply:
Done.